api/pkg/client/mlflow/mlflow.go

```diff
@@ -0,0 +1,188 @@
+package mlflow
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type mlflowClient struct {
+	Client *http.Client
+	Config Config
+}
+
+type Config struct {
+	TrackingURL string
+}
+
+type Mlflow interface {
+	SearchRunForExperiment(idExperiment string) (SearchRunsResponse, error)
+	SearchRunData(idRun string) (SearchRunResponse, error)
+	DeleteExperiment(idExperiment string) error
+	DeleteRun(idRun string) error
+}
+
+type DeleteExperimentRequest struct {
+	ExperimentId string `json:"experiment_id" required:"true"`
+}
+
+type DeleteRunRequest struct {
+	RunId string `json:"run_id" required:"true"`
+}
+
+type DeleteExperimentErrorResponse struct {
+	ErrorCode string `json:"error_code"`
+	Message   string `json:"message"`
+}
+
+type SearchRunRequest struct {
+	ExperimentId []string `json:"experiment_ids" required:"true"`
+}
+
+type TagRun struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+type DataRun struct {
+	Tags []TagRun `json:"tags"`
+}
+
+type InfoRun struct {
+	RunId          string `json:"run_id"`
+	ExperimentId   string `json:"experiment_id"`
+	UserId         string `json:"user_id"`
+	LifecycleStage string `json:"lifecycle_stage"`
+	ArtifactURI    string `json:"artifact_uri"`
+}
+type RunResponse struct {
+	Info InfoRun `json:"info"`
+	Data DataRun `json:"data"`
+}
+type SearchRunsResponse struct {
+	RunsData []RunResponse `json:"runs"`
+}
+type SearchRunResponse struct {
+	RunData RunResponse `json:"run"`
+}
+
+func NewMlflowClient(httpClient *http.Client, config Config) *mlflowClient {
+	return &mlflowClient{
+		Client: httpClient,
+		Config: config,
+	}
+}
+
+func (mfc *mlflowClient) httpCall(method string, url string, headers map[string]string, body []byte, response interface{}) error {
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+
+	resp, err := mfc.Client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if !(resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices) {
+		// Convert response body to Error Message struct
+		var errMessage DeleteExperimentErrorResponse
+		if err := json.NewDecoder(resp.Body).Decode(&errMessage); err != nil {
+			return err
+		}
+		return fmt.Errorf(errMessage.Message)
+	}
+
+	if response != nil {
+		if err := json.NewDecoder(resp.Body).Decode(response); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (mfc *mlflowClient) SearchRunForExperiment(idExperiment string) (SearchRunsResponse, error) {
+	// Search related runs for an experiment id
+	var responseObject SearchRunsResponse
+
+	searchRunURL := fmt.Sprintf("%s/api/2.0/mlflow/runs/search", mfc.Config.TrackingURL)
+
+	headers := map[string]string{
```

Comment:
As we are setting the same header everywhere, can this be added to `httpCall` instead?